Assets/Scripts/Enemies/Armadillo/Behaviour/IBehaviour.cs

```diff
@@ -0,0 +1,7 @@
+namespace Enemies.Armadillo.Behaviour {
+    public interface IBehaviour {
```

Comment:
This and the raycast-related utilities probably belong in a different namespace.